app/models/user.rb

```diff
@@ -507,12 +507,13 @@ def sanitize_role
   def prepare_new_user!
     BootstrapTimelineWorker.perform_async(account_id)
     ActivityTracker.increment('activity:accounts:local')
+    ActivityTracker.record('activity:logins', id)
     UserMailer.welcome(self).deliver_later
     TriggerWebhookWorker.perform_async('account.approved', 'Account', account_id)
   end
 
   def prepare_returning_user!
-    ActivityTracker.record('activity:logins', id)
+    ActivityTracker.record('activity:logins', id) if confirmed?
```

Comment:
We don't need to execute this entire method if neither `confirmed?` nor `approved?` are true.

Reply:
I changed it for a guard clause, but I'm uncomfortable with changing the behavior further wrt. `approved?`.